### DISCUSSION TITLE
TTS+keyboard

### DISCUSSION THREAD

**Discussion by Pantyhose-X** [google/mozc#558] [Ideas]
Hello I have a good idea you can develop a "listen"keyboard tool.
TTS+keyboard
> * [ ] Applied to language learner. Language worker's
> * [ ] Applied to language learning. Pronunciation correction

Learning a foreign language.  Example Russian German I don't know how to speak, TTS taught me to speak
Text-to-Speech Always speak typed keys speak Phonetic letters are spoken when long pressing on keyboard keys and navigating by character.

* example，
https://play.google.com/store/apps/details?id=com.google.android.marvin.talkback

https://user-images.githubusercontent.com/42504385/132130706-a1a8ec10-416a-4711-b5d3-c91086e9710f.mp4

https://www.youtube.com/watch?v=YPBn6D6pf78&t=207s

https://user-images.githubusercontent.com/42504385/127757873-c3c076d3-ce73-40b1-8f57-423459fc0c2f.MP4

* l only want to talk when you press keys on the keyboard, not to read the screen

# Additional context
[Google Text-to-Speech apk](https://play.google.com/store/apps/details?id=com.google.android.tts)

**Comment by hiroyuki-komatsu** [comment]
Hi, thank you for your suggestion.

I guess the users might want to apply the TTS feature not only for keyboards but also for any other texts like contents in web browsers.
So, I think isolation the TTS feature from the keyboards makes sense.

On the other hand, keyboards can provide some annotations for TTS especially for accessibility purpose (e.g. description of homophones). If there are application which can utilize those annotations, we would like to consider it.

Thank you!
